src/main/java/org/jabref/logic/importer/fileformat/RisImporter.java

```diff
@@ -193,25 +198,20 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
                         }
                     } else if ("UR".equals(tag) || "L2".equals(tag) || "LK".equals(tag)) {
                         fields.put(FieldName.URL, value);
-                    } else if (!foundDate && (("Y1".equals(tag) || "Y2".equals(tag) || "PY".equals(tag) || "DA".equals(tag)) && (value.length() >= 4))) {
-                        String year = value.substring(0, 4);
+                    } else if (isDateTag(tag) && value.length() >= 4) {
```

Comment:
I have an even better idea, you can omit one search in the list, e.g. next to the month initialisation
If you directly use the indexOf before the if/else if cases. you can then simply reuse the value if it's <0

Reply:
I coudn't figure out what you meant here, the month initialization is outside of the loop 
```java
            Optional<Month> month = Optional.empty();
            Map<String, String> fields = new HashMap<>();

            String[] lines = entry1.split("\n");

            for (int j = 0; j < lines.length; j++) {
```
but the tag variable is only declared and used inside of the loop, so I couldn't use indexOf(tag) at this point in the code, so I kept the if statement that way.